Components/Adyen/OriginKeysService.php

```diff
@@ -28,17 +32,21 @@ public function __construct(
         ApiFactory $apiFactory,
         ModelManager $models
     ) {
-        $mainShop = $models->getRepository(Shop::class)->findOneBy(['default' => 1]);
-        $this->checkoutUtility = new CheckoutUtility($apiFactory->create($mainShop));
+        $this->apiFactory = $apiFactory;
+        $this->models = $models;
     }
 
     /**
      * @param array $originDomains
      * @return mixed
      * @throws AdyenException
      */
-    public function generate(array $originDomains)
+    public function generate(array $originDomains, Shop $shop)
     {
-        return $this->checkoutUtility->originKeys(['originDomains' => $originDomains])['originKeys'];
+        $checkoutUtility = new CheckoutUtility(
+            $this->apiFactory->provide($shop)
```

Comment:
Client is created or fetched from cache during runtime to make certain the apiClient is created with the correct configuration of the provided shop.